Feature/inba 961 + 986 remove grommet syling

eawest2 opened the pull request:
#### What does this PR do?

Removes the Grommet package, and all remaining references to grommet in the SCSS and JS.

#### Related JIRA tickets:

INBA-961

#### How should this be manually tested?
Fundamentally, it requires a comparative side-by-side test against staging because so many files were changed. Check the following list comprehensively:

PM USER:
1) The All Users menu
2) The User Menu Modal (Accessible by clicking any user) Check all Tabs
3) Flag management screen
4) The Users tab of project management
5) The Users tab of the project creation wizard
6) Moving a user in and out of a staging slot in project management and creation
7) Subjects screen and list
8) All editable text inputs (test for activity, and when you enter an incorrect value or no value)
9) Creating and editing a survey
10) Task details modal (assign a user to a stage and click the ellipses)
11) Date pickers

USER:
1) The user profile (icon at top left)     
2) All editable text inputs
3) User flag screen
4) Survey screen (take a survey)

#### More Instruction

I fully expect I missed some stuff, so please point anything out you find that's off from Staging, and let me know so I can correct.

Commit:
commented out grommet references. Testing now to see effects on documents.

Commit:
removed account-tab grommet styling

Commit:
Removed all grommet references and performed regression against staging.

Commit:
cleared out some commented code

Commit:
removed final reference to grommet in commented code.

Commit:
Took a hatchet to Grommet

eawest2 commented on src/styles/_validated-text-input.scss:
```diff
@@ -1,10 +1,10 @@
-            border-color: red;
+        &--error,
+        input[type=text] {
+            border: 1px solid red !important;
```
@jmfadeley This is the important I had to add.

morleegan commented:
`grep -r grommet indaba-client` shows up clean. so, it looks like all direct references have been removed. GJ.

rileyorr reviewed (changes requested):
**PM USER:**
_The All Users menu_

- There’s a border around First Name & Last Name (screenshot)
![screen shot 2019-01-21 at 1 36 36 pm](https://user-images.githubusercontent.com/46329147/51494971-aefbb500-1d88-11e9-8fb8-9d3cc2668709.png)
- There’s no Resend Invitation button
- Notify Level & Status dropdowns have different styling compared to staging
- Account & Preference tabs need to be left aligned
- Should be a space in “Title (Optional,...)” this is incorrect in staging too

_The User Menu Modal (Accessible by clicking any user) Check all Tabs_

- LGTM

_Flag management screen_

- LGTM

_The Users tab of project management_

- LGTM

_The Users tab of the project creation wizard_

- LGTM

_Moving a user in and out of a staging slot in project management and creation_

- The filter search bar above “Filter by groups” is styled differently than in staging. The textbox is smaller and the search icon is outside the textbox.

_Subjects screen and list_

- Modal for deleting a subject doesn’t let the user cancel the action (this is also broken in staging).

_All editable text inputs (test for activity, and when you enter an incorrect value or no value)_

- LGTM unless noted elsewhere

_Creating and editing a survey_

- Increased padding on the question types looks good, but the tab headers need to be left aligned (screenshot)
![screen shot 2019-01-21 at 12 53 23 pm](https://user-images.githubusercontent.com/46329147/51495059-fd10b880-1d88-11e9-93df-39b9ed84c332.png)
_Task details modal (assign a user to a stage and click the ellipses)_

- Reassigning to a new user drop down doesn’t work & reassign function doesn't work 
- Force complete doesn’t show "Done" label unless I refresh the page

_Date pickers_

- LGTM

**USER:**
_The user profile (icon at top left)_

- Notify Level dropdown has a red x in it (and the styling is different from staging)
- Did we get rid of the reset password feature?

_All editable text inputs_

- “Search for Tasks” search bar does not have the same styling & the search icon is outside the textbox


_User flag screen_

- LGTM

_Survey screen (take a survey)_

- LGTM

Arghast commented:
@eawest2 Just FYI, the broken search icons were already an issue throughout the app in IE before this PR. I will report that as a bug if it's outside of the scope.

eawest2 commented:
@rileyorr @Arghast 

Go ahead and tally up the bugs you've found that are present both in this and in staging so once it gets merged we have a list of some styling bugs we can work towards correcting. As of now for the scope of this ticket, I'm just trying to match staging as best as we can using Vanilla styles.

Commit:
partially completed fixes.

Commit:
pushing further changes

Commit:
content push to pr with comment.

eawest2 commented:
As a note, the scope of this ticket is only in styling and SCSS, as none of the components were changed by this PR except in styling and grommet references. As a result, code bugs will need to be addressed in a different ticket. 

@rileyorr 

**All user's menu text fields border:**
This is due to a change in how the text of all the fillable text-fields work on the site with Jame's INBA-971. Without refactoring every component they are in (which may need to be done anyway later on) there's no way to fix this without adding in bad-practice code. Basically, when Grommet was applying CSS settings, it was overriding everything else going on in that element. The code isn't written that way anymore, so unless I get the go-ahead from James to add in an !important every time we have a grommet overwrite, that's not going to change in the scope of this ticket. 

**Resend invitation button:**

You may need to add another user who's not accepted, I've got one on my local explicitly for this reason, and I got the button to show up. 
![image](https://user-images.githubusercontent.com/35747608/51557276-73292400-1e4a-11e9-850d-f85ddaf66a2d.png)

**Dropdown Styling:** Fixed to add box highlights and prevent jitter of page elements due to changing padding/border values. 

**Alignment of Tabs Left:** Fixed inherited padding from Chrome User Agent Stylesheet. We should probably find out why it's applying styles, since it's not supposed to be. 

**Title options typo:** Added space onto "Title" in i18n. Fixed issue.

**Staging Slot:** Fixed staging slot styling as best as is possible. Revealed some larger issues with core styling changes with grommet removal. 

**Subject Screen and list:** As this is broken in staging, it'll probably need to be a bug ticket, rather than addressed here. @rileyorr, go ahead and make one in Jira for it.

**Creating and editing a survey:** 
1)Removed inline start padding from tabs
2) IIRC this is an existing bug that has been around since the modal was developed. Check to make sure a ticket doesn't already exist for the problem (I think one does for this sprint. Ask in Slack and check Jira) and if not, create it in Jira.
3) This is probably a React issue, and will have to be corrected in code rather than styles.

**User Profile:**
1)I don't see this red X on my end. It may be the same artifact you had with your users earlier. Try creating a new user and trying that with the new user. The styling seems more or less the same as staging.
2) This functionality is removed since the button didn't actually do anything, and the intended functionality was outside of best practice. At some point we'll replace it with working functionality.

**Search for Tasks:**
This looks good for my end. I'm not sure what the difference might be. 

@Arghast 

**Magnifying glass:**
I fixed this in line with Riley's comments. 

**Survey Question Date:**
It seems that this is part of the established functionality of the component, however it's unusual in that it cuts off before the end of the question box. I wasn't able to find a fix for this.

rileyorr commented:
@eawest2 
**Resend Button Invitation**
Got this to work - thanks!

**Dropdown Styling**
Styling looks good! However, when I'm testing by completing a survey, any option I click on results in a blank screen. Seems unrelated to styling, though.

**Alignment of Tabs Left**
LGTM! 

**Title Options Typo**
LGTM!

**Staging Slot**
LGTM!

**Subject screen & list**
It looks like I'm seeing the cancel button on my end now, so I'm not sure what was going on initially.

**Creating & Editing a Survey**
Added a ticket!

**User Profile**
I might be looking at the wrong thing, but this is what I'm seeing:
![screen shot 2019-01-25 at 11 29 43 am](https://user-images.githubusercontent.com/46329147/51758802-90e9c980-2094-11e9-80cd-7318a21c9642.png)

**Search for Tasks**
LGTM!

jmfadeley committed:
Merge branch 'develop' into feature/inba-961-remove-grommet-syling

Merging develop into 961.

jmfadeley committed:
[INBA-961] Minor improvements for the search input.

jmfadeley committed:
[INBA-961] Improvement for the editable text input styling.

jmfadeley committed:
[INBA-961] The em units for css are very different between IE and
Chrome, at least for height. As such, I've given the text area in the
Task Options Form a minimum height of 80px.

jmfadeley committed:
[INBA-961][INBA-986] Update that will free us from needing to use
!important and allow for specific styling of modal inputs again.

jmfadeley committed:
[INBA-961] Styling tweaks.

jmfadeley committed:
[INBA-961] Removal of all !importants.

jmfadeley committed:
[INBA-961] Have to roll back one !important removal.

jmfadeley committed:
[INBA-961] Fix for the appearance of the status cards.

jmfadeley commented:
@eawest2 @Arghast I need you two to take a look at this in IE. I've combined this PR with INBA-986, which includes refactoring to allow modal text inputs to be controllable again, and many, many minor fixes and tweaks in the appearance including (of course), Internet Explorer. @rileyorr I don't think you can test it unless they have IE for a Mac, but if you can, feel free to take a look.

jmfadeley committed:
[INBA-961][INBA-986] Forgot the hover.

morleegan commented on src/styles/_preference-tab.scss:
```diff
@@ -24,6 +24,11 @@ $block-class:'preference-tab';
+            &:focus {
+                padding: 9px 21px;
+                border: 2px solid #4eb276;
+            }
```
Hey Eric, this code `input-select` and `preference tab input-textarea`. is the same code, you could take it out and make it into a `@mixin` so that the code is shared between the two.

Additionally, we want to use a variable name for this color hex which I believe is `brand-name`. located in styles.scss

morleegan commented on src/styles/_preference-tab.scss:
```diff
@@ -39,6 +44,11 @@ $block-class:'preference-tab';
+
+            &:focus {
+                padding: 9px 21px;
+                border: 2px solid #4eb276;
```
look above about color and sharing code.

morleegan commented on src/styles/dashboard/_filter-input.scss:
```diff
@@ -12,32 +12,35 @@ $block-class: 'filter-input';
         }
 
+        &__input {
+            color: #909da7;
```
same thing here with color hex codes.

morleegan reviewed (commented):
I think it is important to fix the rampant random color codes around the scss. in the files you made changes to make sure that the hex's are variables in styles.

morleegan commented:
@eawest2 - this branch now has merge conflicts

jmfadeley commented:
@eawest2 @morleegan We'll take a look on Wednesday. This is the ticket that is not in the sprint, so Kate wants it left out for now. I will handle the conflicts and the color issues you pointed out then.

jmfadeley committed:
Merging of develop.

jmfadeley committed:
Merge branch 'develop' into feature/inba-961-remove-grommet-syling

secondary merge of develop.

jmfadeley committed:
Last time with lint fixes.

jmfadeley committed:
[INBA-961]  Change for weird $background-color variable I cannot account
for.

jmfadeley commented on src/styles/common/_react-select-override.scss:
```diff
@@ -4,7 +4,7 @@
     .Select--multi .Select-value {
         color: $form-text;
-        background-color: $background-color;
+        background-color: #fff;
```
So this is a mystery I can't explain. This variable was throwing an error in this branch, but not `develop`. Yet I've gone through references on `develop` and can't seem to find reference to this variable. For the time being I've subbed it out for plain, triple-f white. If @morleegan can explain where this is coming from, that would be great.

jmfadeley committed:
[INBA-961] Changes per Morgan's PR comments.

jmfadeley committed:
Corrected to LIGHT_GRAY for the react select.

jmfadeley committed:
[INBA-961] Added a $WHITE css variable.

jmfadeley commented:
@eawest2 @Arghast you guys are clear to test. @rileyorr you're welcome if you've time but I know you're busy today.

eawest2 commented:
**Method**

Brought down the latest changes and develop. Fired up the branch micro-services, and examined in chrome.

**Testing**
Performed the tests I laid out in the initial post. The only issue I found is the following:

Dropdowns in all users / user profile / preference tab still have blue box highlights a-la browser default.

This seems to have propagated over to staging, however I'm fairly certain it wasn't this way originally. @morleegan @laceyIrvin do either of you know if it was different originally?

**Conclusion**

The PR seems good to merge, apart from the one issue I found, which is already live anyway.

Arghast commented:
@jmfadeley LGTM Tested in Chrome, Firefox, IE and Edge.

jmfadeley commented:
@eawest2 @Arghast alright, record the issue and merge it. Don't forget to set the ticket as "In Testing."
